tasks/main.yml

```diff
@@ -11,7 +11,12 @@
 - name: pip install monasca-agent in a virtualenv
   pip: name=monasca-agent state=latest virtualenv="{{monasca_virtualenv_dir}}"
   notify: run monasca-setup
-  when: not skip_install
+  when: not skip_install and monasca_agent_version is not defined
+
+- name: pip install monasca-agent in a virtualenv
```

Comment:
It seems confusing these two tasks have the same name. You will get one that runs and one that is skipped